sbt-dotty/src/dotty/tools/sbtplugin/DottyIDEPlugin.scala

```diff
@@ -89,40 +168,40 @@ object DottyIDEPlugin extends AutoPlugin {
   )
 
   override def buildSettings: Seq[Setting[_]] = Seq(
-    configureIDE := {
-      val log = streams.value.log
-
-      val configs0 = state.flatMap(s =>
-        inAllDottyConfigurations(projectConfig, s)
-      ).value.flatten
-      // Drop configurations who do not define their own sources, but just
-      // inherit their sources from some other configuration.
-      val configs = distinctBy(configs0)(_.sourceDirectories.deep)
-
-      if (configs.isEmpty) {
-        log.error("No Dotty project detected")
-      } else {
-        // If different versions of Dotty are used by subprojects, choose the latest one
-        // FIXME: use a proper version number Ordering that knows that "0.1.1-M1" < "0.1.1"
-        val ideVersion = configs.map(_.compilerVersion).sorted.last
+    configureIDE := Def.taskDyn {
+      val origState = state.value
+      Def.task {
+        val (dottyVersion, projRefs, dottyState) = dottySetup(origState)
+        val configs0 = runInAllConfigurations(projectConfig, projRefs, dottyState).flatten
+
+        // Drop configurations who do not define their own sources, but just
+        // inherit their sources from some other configuration.
+        val configs = distinctBy(configs0)(_.sourceDirectories.deep)
+
         // Write the version of the Dotty Language Server to use in a file by itself.
         // This could be a field in the JSON config file, but that would require all
         // IDE plugins to parse JSON.
+        val dlsVersion = dottyVersion
+          .replace("-nonbootstrapped", "") // The language server is only published bootstrapped
+        val dlsBinaryVersion = dlsVersion.split("\\.").take(2).mkString(".")
         val pwArtifact = new PrintWriter(".dotty-ide-artifact")
-        pwArtifact.println(s"ch.epfl.lamp:dotty-language-server_0.1:${ideVersion}")
+        pwArtifact.println(s"ch.epfl.lamp:dotty-language-server_${dlsBinaryVersion}:${dlsVersion}")
         pwArtifact.close()
```

Comment:
Idk, visceral reaction to seeing an allocation of a file stream and its `close()` without `try/finally`. Right now it might not make a difference, but who knows what will happen in the future with that code, and someone will introduce a potentially-exception-throwing code and forget to add the try/finally.